File: week1/Arrays/Level2/MajorityElement.java
import java.util.Scanner;

public class MajorityElement {
   public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int arr[] = new int[n];
        int count = 0;
        int ele = -1;
        for(int i=0;i<n;i++){
            arr[i] = sc.nextInt(); 
            if(count==0){
                ele = arr[i];
                count++;
            }
            else if(arr[i]==ele) count++;
            else count--;
        }
        System.out.println(ele);
   }
}

File: week1/Arrays/Level2/MajorityElement.py
from typing import List
def majorityElement(nums: List[int]) -> int:
    count = ele = 0
    for num in nums:
        if count == 0:
            ele = num
        count += 1 if ele == num else -1
    return ele

nums = list(map(int,input().split()))
print(majorityElement(nums))